### PR TITLE
Add Gist API

### PR DESCRIPTION
**Pull request by aeei**
# Add Gist API

### Prepare list before merge this commit
- Setting redis for test
- Create Github OAuth app
- Share .env file include redis & Github OAuth app (client_id/client_secret/callback_url) information

### Added modules
- redis & @types
- connect-redis & @types
- express-session & @types
- cookie-parser & @types
- passport & @types
- passport-github & @types
- request & @types
- uuid & @types

### Gist API
Action | HTTP request method | API URL | Specification
:-- | :-- | :-- | :--
Get user's gists| GET | `/gists` | [link](https://developer.github.com/v3/gists/#list-a-users-gists)
Get user's gist | GET | `/gists/:gist_id` | [link](https://developer.github.com/v3/gists/#get-a-single-gist)
Create gist | POST | `/gists` | [link](https://developer.github.com/v3/gists/#create-a-gist)
Modify gist | PATCH | `/gists/:gist_id` | [link](https://developer.github.com/v3/gists/#edit-a-gist)
Delete gist | DELETE | `/gists/:gist_id` | [link](https://developer.github.com/v3/gists/#delete-a-gist)

### Login with Github OAuth
Action | HTTP request method | API URL | Specification
:-- | :-- | :-- | :--
Login | GET | `/auth/github` | [link](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#1-users-are-redirected-to-request-their-github-identity)

### CODE REVIEW

**Comment by gimdongwoo** [comment]
Awesome!

File: server/handlers/gist.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import request from 'request';
+import { name, version } from './../../package.json';
+
+const getGists = (req: Request, res: Response, next: NextFunction) => {
+  console.log(req.user);
+  request.get(
+    `https://api.github.com/users/${req.user.username}/gists`,
+    {
+      json: true,
+      headers: {
```
Comment (gimdongwoo) [inline server/handlers/gist.ts]: We need to use common authorize token from environment variables for Github's request limit.

File: server/handlers/gist.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      res.send(body);
+    },
+  );
+};
+const getGist = (req: Request, res: Response, next: NextFunction) => {
+  request.get(
+    `https://api.github.com/gists/${req.params.gist_id}`,
+    {
+      json: true,
+      headers: {
```
Comment (gimdongwoo) [inline server/handlers/gist.ts]: We need to use common authorize token from environment variables for Github's request limit.

File: server/handlers/gist.ts (hunk 1 above)
Comment (aeei) [reply server/handlers/gist.ts]: Oh, I see !!
I will add common authorize token into code and then pull request again.

**Comment by aeei** [comment]
Required common authorize token.

**Pull request closed without merging**